traffic_ops/traffic_ops_golang/cdnnotification/cdnnotifications.go

```diff
@@ -0,0 +1,190 @@
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+	"net/http"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+)
+
+const readQuery = `
+SELECT cn.cdn, 
+	cn.last_updated,
+	cn.user, 
+	cn.notification 
+FROM cdn_notification as cn
+INNER JOIN cdn ON cdn.name = cn.cdn
+INNER JOIN tm_user ON tm_user.username = cn.user
+`
+
+const insertQuery = `
+INSERT INTO cdn_notification (cdn, "user", notification)
+VALUES ($1, $2, $3)
+RETURNING cdn_notification.cdn,
+          cdn_notification.last_updated,
+          cdn_notification.user,
+          cdn_notification.notification
+`
+
+const deleteQuery = `
+DELETE FROM cdn_notification
+WHERE cdn_notification.cdn = $1
+RETURNING cdn_notification.cdn,
+          cdn_notification.last_updated,
+          cdn_notification.user,
+          cdn_notification.notification
+`
+
+// Read is the handler for GET requests to /cdn_notifications.
+func Read(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	cdnNotifications := []tc.CDNNotification{}
+
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"cdn":  dbhelpers.WhereColumnInfo{"cdn.name", nil},
+		"user": dbhelpers.WhereColumnInfo{"tm_user.username", nil},
+	}
+
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(inf.Params, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		sysErr = util.JoinErrs(errs)
+		errCode = http.StatusBadRequest
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	query := readQuery + where + orderBy + pagination
+	rows, err := inf.Tx.NamedQuery(query, queryValues)
+	if err != nil {
+		userErr, sysErr, errCode = api.ParseDBError(err)
+		if sysErr != nil {
+			sysErr = fmt.Errorf("notification read query: %v", sysErr)
+		}
+
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var n tc.CDNNotification
+		if err = rows.Scan(&n.CDN, &n.LastUpdated, &n.User, &n.Notification); err != nil {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("scanning cdn notifications: "+err.Error()))
+			return
+		}
+		cdnNotifications = append(cdnNotifications, n)
+	}
+
+	api.WriteResp(w, r, cdnNotifications)
+}
+
+// Create is the handler for POST requests to /cdn_notifications.
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if sysErr != nil || userErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var n tc.CDNNotification
```

Comment:
Shouldn't this be a `tc.CDNNotificationRequest`? I mean, I think this works since you're using basic `QueryRow`, but I just assume this is the reason you made that struct.

Reply:
done